### PR TITLE
Fixes mod paint kit

### LINKED ISSUE #84490: Paint kit right click doesn't work on Modsuits with storage
## Issue Summary

See title, paint kits are unable to recolour modsuits via right click, if the modsuit has storage

## Round ID:

Issue report from https://github.com/Bubberstation/Bubberstation

## Reproduction:

This seems to be due to a change after combat mode was made more needed to use tools on modsuits.
Paint kits can change modsuit skins (left click) but right click does nothing if the modsuit has a storage module (it just opens storage)

### LINKED ISSUE #84620: Mod paint kit fails
## Issue Summary

Mod paint kit cant re-skin or repaint modsuits. Which was based on left and right click respectively. This is happening despite the previous fix pr mentioned in #83442 

This seems to be happening due to the combat mode and dehardcoding of modsuits.

## Round ID:

NA

## Testmerges:

NA(Happens Locally as well)

## Reproduction:

1.  Print Modsuit paint kit
2. Try left clicking on modsuit, try right clicking, neither works.
3. Repeat step 2 with combat mode & Alt(see my run/walk change) on. Nothing.
4. Test on other modsuits. Nothing.

https://github.com/tgstation/tgstation/assets/68121607/13cc834b-fda2-4f33-b38c-1bcb9dc6cfd6

### PR DESCRIPTION
**Pull request by GPeckman**
## About The Pull Request

After all the attack chain refactors, the modsuit paint kit was bugged. If you tried to use it on a modsuit that had storage installed, then the paint kit would simply be inserted into the modsuit instead of allowing you to change the skin/color.

Fixes #84620 
Fixes #84490 
## Changelog
:cl:
fix: The modsuit paint kit is no longer broken.
/:cl:

### CODE REVIEW

**Review by MrMelbert — APPROVED** [review]
This fix is OK but we should still find a better solution to this at some point

**Comment by GPeckman** [comment]
I'm curious what a better solution would look like? The docs for `tool_act` list overriding the target's storage as one of the actual use cases for the proc

**Comment by MrMelbert** [comment]
One does not exist at the moment which is why I said we should find one

File: code/modules/mod/mod_control.dm [hunk 1]
```diff
@@ -322,6 +322,29 @@
...
+
+			paint_kit.proxy_view.appearance = paint_kit.editing_mod.appearance
+			paint_kit.proxy_view.color = null
+			paint_kit.proxy_view.display_to(user)
+			paint_kit.ui_interact(user)
+			return ITEM_INTERACT_SUCCESS
+		else // Left click
+			if(active || activating)
+				balloon_alert(user, "suit is active!")
+				return ITEM_INTERACT_BLOCKING
```
Comment (Ghommie) [inline code/modules/mod/mod_control.dm:345]:
```suggestion
```

File: code/modules/mod/mod_control.dm [hunk 2]
```diff
@@ -322,6 +322,29 @@
 			return ITEM_INTERACT_BLOCKING
 		insert_pai(user, tool)
 		return ITEM_INTERACT_SUCCESS
+	if(istype(tool, /obj/item/mod/paint))
+		var/obj/item/mod/paint/paint_kit = tool
+		if(LAZYACCESS(modifiers, RIGHT_CLICK)) // Right click
+			if(active || activating)
+				balloon_alert(user, "suit is active!")
+				return ITEM_INTERACT_BLOCKING
```
Comment (Ghommie) [inline code/modules/mod/mod_control.dm:330]:
```suggestion
		if(active || activating)
			balloon_alert(user, "suit is active!")
			return ITEM_INTERACT_BLOCKING
		if(LAZYACCESS(modifiers, RIGHT_CLICK)) // Right click
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Pull request merged**